### DISCUSSION TITLE
Super cluster [Core NATS] with internal subjects

### DISCUSSION THREAD

**Discussion by mrForest13** [nats-io/nats-server#4746] [Q&A] [answered]
Hello,

We have two cluster in our infrastructure, one in Europe and one in Asia. It's a super cluster. I wonder if we can mark some topics to be internal in a local cluster? So for example:

Eu Cluster
- external.a
- external.b
- internal.a

Asia Cluster
- external.c
- external.d
- internal.a

So as a result we want to share external topic between cluster, but internal should be accessible only for local subscriptions. We have one requirement: internal topics have the same name in each cluster

Is it possible? Probably it can be done by users and permissions, but i am not sure if it's possible with the same subjects names

**Comment by derekcollison** [comment, marked as answer]
If you want to more strictly control traffic between clusters, leafnodes may be a better fit here with explicit permissions set on the leafnode connections that bind the local and remote accounts.

**Reply by mrForest13** [reply]
ok, it makes sense. What about just using accounts? So in my case:

EU cluster account with exports only external topics and and same for Asia cluster so they will not be able to see internal topics in super cluster?

**Reply by derekcollison** [reply]
Right now there is not a clean way to say an account can not be loaded into a system. If you are using resolvers you could limit it that way, and we do have certain abilities with CIDR blocks. But easiest would be to leafnode between EU and APAC and explicitly control subjects on the leafnode itself.
